src/java/org/apache/cassandra/index/sai/disk/v1/SSTableIndexWriter.java

```diff
@@ -62,11 +62,13 @@ public class SSTableIndexWriter implements PerColumnIndexWriter
     public static final long MAX_STRING_TERM_SIZE = SAI_MAX_STRING_TERM_SIZE.getSizeInBytes();
     public static final long MAX_FROZEN_TERM_SIZE = SAI_MAX_FROZEN_TERM_SIZE.getSizeInBytes();
     public static final long MAX_VECTOR_TERM_SIZE = SAI_MAX_VECTOR_TERM_SIZE.getSizeInBytes();
-    public static final String TERM_OVERSIZE_MESSAGE = "Can't add term of column {} to index for key: {}, term size {} " +
+    public static final String TERM_OVERSIZE_MESSAGE = "Can't add term of column {} to index for: {}, term size {} " +
                                                        "max allowed size {}, use analyzed = true (if not yet set) for that column.";
 
     private final IndexDescriptor indexDescriptor;
-    private final IndexContext indexContext;
+    private final StorageAttachedIndex index;
+    private final IndexTermType indexTermType;
+    private final IndexIdentifier indexIdentifier;
```

Comment:
Maybe we should choose between storing a reference to the `StorageAttachedIndex` and storing references to its relevant components (`IndexTermType`, `IndexIdentifier`, `AbstractAnalyzer`, `IndexMetrics` and `IndexWriterConfig`). Having references to both the container and its contents might feel redundant, wdyt?

Reply:
Agreed. I have removed storing the reference to the index and am storing the `IndexMetrics` and the `IndexWriterConfig` instead. I also removed the null checks on the index metrics in the code because the `StorageAttachedIndex` always creates the `IndexMetrics` now. The null check was a throwback to the `IndexContext`.